crates/ruma-events/src/direct.rs

```diff
@@ -81,19 +197,48 @@ mod tests {
     #[test]
     fn deserialization() {
         let alice = owned_user_id!("@alice:ruma.io");
+        let alice_mail = "alice@ruma.io";
         let rooms = vec![owned_room_id!("!1:ruma.io"), owned_room_id!("!2:ruma.io")];
+        let mail_rooms = vec![owned_room_id!("!3:ruma.io")];
 
         let json_data = json!({
             "content": {
-                alice.to_string(): rooms,
+                alice.clone(): rooms,
+                alice_mail: mail_rooms,
             },
             "type": "m.direct"
         });
 
         let event: DirectEvent = from_json_value(json_data).unwrap();
-        let direct_rooms = event.content.get(&alice).unwrap();
 
+        let direct_rooms = event.content.get(&OwnedDirectUserIdentifier::from(alice)).unwrap();
```

Comment:
To avoid an allocation here and below, you can use:
```suggestion
        let direct_rooms = event.content.get(<&DirectUserIdentifier>::from(alice)).unwrap();
```